extensions/openshift-ai/src/openshiftai.ts

```diff
@@ -0,0 +1,299 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+import { createOpenAICompatible } from '@ai-sdk/openai-compatible';
+import type {
+  Disposable,
+  Provider,
+  provider as ProviderAPI,
+  ProviderConnectionStatus,
+  SecretStorage,
+} from '@kortex-app/api';
+import { CoreV1Api, CustomObjectsApi, KubeConfig } from '@kubernetes/client-node';
+
+export const TOKENS_KEY = 'openshiftai:infos';
+export const TOKEN_SEPARATOR = ',';
+const INFO_SEPARATOR = '|';
+
+interface ConnectionInfo {
+  token: string;
+  baseURL: string;
+}
+
+export class OpenShiftAI implements Disposable {
+  private provider: Provider | undefined = undefined;
+  private connections: Map<ConnectionInfo, Disposable> = new Map();
+
+  constructor(
+    private readonly providerAPI: typeof ProviderAPI,
+    private readonly secrets: SecretStorage,
+  ) {}
+
+  async init(): Promise<void> {
+    // create provider
+    this.provider = this.providerAPI.createProvider({
+      name: 'OpenShift AI',
+      status: 'unknown',
+      id: 'openshiftai',
+      emptyConnectionMarkdownDescription: 'Provides OpenShift AI integration. Connects Kortex to models running on OpenShift AI.',
+      images: {
+        icon: './icon.png',
+        logo: {
+          dark: './icon.png',
+          light: './icon.png',
+        },
+      },
+    });
+
+    // register inference Provider connection factory
+    this.provider?.setInferenceProviderConnectionFactory({ create: this.inferenceFactory.bind(this) });
+
+    // restore persistent connections
+    await this.restoreConnections();
+  }
+
+  private async restoreConnections(): Promise<void> {
+    const connectionInfos = await this.getConnectionInfos();
+    for (const connectionInfo of connectionInfos) {
+      try {
+        await this.registerInferenceProviderConnection({
```

Comment:
suggestion:         
await this.registerInferenceProviderConnection(connectionInfo);